### ISSUE TITLE
octave asks for TeX when building from source.

### ISSUE THREAD

**Issue by xu-cheng** [Homebrew/homebrew-science#2069]
@iml Let's move to a new issue to fix the TeX problem. This is the gist-logs: https://gist.github.com/xu-cheng/594793edb7f62d0fc8bc (I truncated compiling part of log as it's huge and not related.)

**Comment by dpo** [comment]
Just for info, according to `INSTALL.OCTAVE`:
````
     '--disable-docs'
          Disable building all forms of the documentation (Info, PDF,
          HTML). The default is to build documentation, but your system
          will need functioning Texinfo and TeX installs for this to
          succeed.
````
but I haven't tried on a machine without TeX installed.

**Comment by xu-cheng** [comment]
While I have MacTeX installed. But since you remove `depends_on :tex`, Homebrew's superenv will prevent octave find it.

**Comment by dpo** [comment]
Actually I also have MacTeX installed, but `brew install octave` completes without trouble.

**Comment by xu-cheng** [comment]
Did you build from source?

**Comment by dpo** [comment]
Always do.

**Comment by xu-cheng** [comment]
Judging by the same outcome of CI bot. My Mac has some settings interferencing the build process. I'm away from my computer now, so I may look it tomorrow. My first guess is the `.latexmkrc` file.

**Comment by dpo** [comment]
@xu-cheng Were you able to resolve this?

**Comment by xu-cheng** [comment]
No, I haven't resolved it yet. And didn't do more digging on that as I need to use octave and bottle is available.

**Comment by dpo** [comment]
It's probably safe to close this then; the issue was probably local.

**Comment by yihangho** [comment]
I think I'm having the same issue - installing Octave with `build-from-source` option on is failing. Here's the [gist-logs](https://gist.github.com/anonymous/1efd90e12f840fe5c1ac) (take note especially of [03.make](https://gist.githubusercontent.com/anonymous/1efd90e12f840fe5c1ac/raw/03.make)). And yes, I do have TeX installed and PATH correctly configured (`which tex` says `/usr/texbin/tex`).

**Comment by tdsmith** [comment]
The error's coming from texi2dvi. Can you start the octave build with `--debug`, interrupt it, and then try to run `which tex`, `which texi2dvi`, and run a file through texi2dvi to see what it does?

**Comment by yihangho** [comment]
```
octave-3.8.2 $ which tex
/usr/texbin/tex
octave-3.8.2 $ which texi2dvi
/usr/bin/texi2dvi
octave-3.8.2 $ cd doc
doc $ texi2dvi interpreter/intro.texi
sed: 2: "s/\(^\|.* \)@documenten ...": whitespace after branch
sed: 4: "s/\(^\|.* \)@documenten ...": whitespace after label
sed: 6: "s/\(^\|.* \)@documenten ...": undefined label 'found                                                '
This is pdfTeX, Version 3.14159265-2.6-1.40.15 (TeX Live 2014) (preloaded format=etex)
 restricted \write18 enabled.
entering extended mode

(/private/tmp/octave20150531-2456-mctnc7/octave-3.8.2/doc/interpreter/intro.tex
i
/private/tmp/octave20150531-2456-mctnc7/octave-3.8.2/doc/interpreter/intro.texi
:92: Missing $ inserted.
<inserted text> 
                $
<to be read again> 
                   ^
l.92 ...ic operations (+,-,*,/), exponentiation (^
                                                  ),
? 
doc $ texi2dvi interpreter/octave.texi 
sed: 2: "s/\(^\|.* \)@documenten ...": whitespace after branch
sed: 4: "s/\(^\|.* \)@documenten ...": whitespace after label
sed: 6: "s/\(^\|.* \)@documenten ...": undefined label 'found                                                '
This is pdfTeX, Version 3.14159265-2.6-1.40.15 (TeX Live 2014) (preloaded format=etex)
 restricted \write18 enabled.
entering extended mode

(/private/tmp/octave20150531-2456-mctnc7/octave-3.8.2/doc/interpreter/octave.te
xi (./texinfo.tex Loading texinfo [version 2009-04-06.11]: pdf, fonts,
page headings, tables, conditionals, indexing, sectioning, toc, environments,
defuns, macros, cross references, insertions, localization, formatting,
and turning on texinfo input format.)
(/private/tmp/octave20150531-2456-mctnc7/octave-3.8.2/doc/interpreter/macros.te
xi)
(/private/tmp/octave20150531-2456-mctnc7/octave-3.8.2/doc/interpreter/version.t
exi)
/private/tmp/octave20150531-2456-mctnc7/octave-3.8.2/doc/interpreter/octave.tex
i:97: epsf.tex not found, images will be ignored.
@image ...f.tex not found, images will be ignored}
                                                  @global @warnednoepsftrue ...
l.97 @flushright @image{octave_logo,2.1in}
                                          
? 
```

**Comment by xu-cheng** [comment]
I just notice `octave` should never `depends_on "texinfo"` as it is already shipped with OS X.

**Comment by guoxiao** [comment]
Is there any progress about this issue? I met with the same problem.

**Comment by dpo** [comment]
Is this still relevant now that 2fce985f0994ada92ae02889f03c43655cbf9927 has been merged?

**Comment by iml** [comment]
Fixed in #2340.
